### PR TITLE
Task/elliottyates/tlt 924/ip modal form validation and submit logic fixes

### PR DESCRIPTION
**Pull request by elliottyates**
- fixes double-click submission issues when reopening the form; also disables Cancel button on submit
- can no longer update or create an IP without valid URLs for all tracks
- switching URL inputs to text inputs fixes issue where built-in HTML5 validation didn't match our own JS validation
- create IP modal now resets every time it is opened
- edit IP modal now reinstates saved values every time it is opened

### CODE REVIEW

**Comment by elliottyates** [comment]
Would love to get @roderickmorales 's eyes on this, plus perhaps one other dev. Also @mmtchong what are we doing on tests for JS & HTML at this point?

File: ab_tool/templates/ab_tool/modals.html [hunk 1]
```diff
@@ -175,6 +181,11 @@ <h4 class="modal-title" id="myModalLabel">Edit Intervention Point</h4>
                         <label for="is_canvas_page_for_track_{{track_url.track.id}}">Open</label>
                         <div class="custom-dropdown">
                             <select name="is_canvas_page_for_track_{{track_url.track.id}}"
+                                data-initial-index="
+                                  {% if not track_url.open_as_tab and track_url.is_canvas_page %}0{% endif %}
```
Comment (douglashall) [inline ab_tool/templates/ab_tool/modals.html]: Is there a way to move this logic into the view?

File: ab_tool/templates/ab_tool/modals.html [hunk 2]
```diff
@@ -175,6 +181,11 @@ <h4 class="modal-title" id="myModalLabel">Edit Intervention Point</h4>
...
+                                  {% if not track_url.open_as_tab and not track_url.is_canvas_page %}1{% endif %}
+                                  {% if track_url.open_as_tab %}2{% endif %}
+                                "
```
Comment (douglashall) [inline ab_tool/templates/ab_tool/modals.html]: Is there a way to move this logic into the view?

File: ab_tool/static/ab_tool/js/experiments_dashboard.js [hunk 3]
```diff
@@ -21,73 +67,53 @@ $(document).ready(function(){
...
+            // disables create/update button, cancel button, and marks form for submit
+            // (so that subsequent calls to enable_submit_if_no_errors will not re-enable form)
+            $submit.addClass("disabled submitting");
+            $submit.siblings('.modal-cancel').addClass("disabled");
+        }
+    });
+
+    $('.intervention-url').on('input', function(){
+        validate_url_input_element($(this));
+        enable_submit_if_no_errors($(this).closest('.modal-form'));
```
Comment (douglashall) [inline ab_tool/static/ab_tool/js/experiments_dashboard.js]: Not that you should fix all of these now, but we should stick to camel-case in Javascript.

Comment (roderickmorales) [inline ab_tool/static/ab_tool/js/experiments_dashboard.js]: +1

**Comment by roderickmorales** [comment]
The notes textarea doesn't get reset when canceling/closing the modal window

**Comment by roderickmorales** [comment]
When you create a new IP, the "preview" link is visible without any URL in the input field. As far as I know, the preview link only should appear if the URL is valid.

**Comment by mmtchong** [comment]
@elliottyates there are some Jasmine unit tests for some of the Angular that I am not as familiar with. Selenium is used to install the tool into a course and create an experiment, and the tests assert that the dashboard is visible to a teacher and that the content of an intervention point is visible to a student.

**Comment by elliottyates** [comment]
Thanks all for the comments, I'll address them and refresh the PR.

**Comment by elliottyates** [comment]
@douglashall -- remaining issue of the logic in the view: I looked into this and each thing I tried (using a mapping dict in the template context, creating a custom template tag, changing the view objects to dicts so I could easily add a new kvp in the appropriately nested position) got hairy quickly; the option of updating the model itself doesn't seem great either (for MVC separation reasons). Given time constraints, I'll msg you offline for options, thanks.

**Comment by douglashall** [comment]
Looks good to me.

**Pull request merged**